Validation/GlobalRecHits/src/GlobalRecHitsAnalyzer.cc

```diff
@@ -28,9 +30,17 @@ GlobalRecHitsAnalyzer::GlobalRecHitsAnalyzer(const edm::ParameterSet& iPSet)
       cscGeomToken_(esConsumes()),
       rpcGeomToken_(esConsumes()),
       count(0) {
-  consumesMany<edm::SortedCollection<HBHERecHit, edm::StrictWeakOrdering<HBHERecHit>>>();
-  consumesMany<edm::SortedCollection<HFRecHit, edm::StrictWeakOrdering<HFRecHit>>>();
-  consumesMany<edm::SortedCollection<HORecHit, edm::StrictWeakOrdering<HORecHit>>>();
+  HBHERecHitgetter_ = edm::GetterOfProducts<edm::SortedCollection<HBHERecHit, edm::StrictWeakOrdering<HBHERecHit>>>(
+      edm::ProcessMatch("*"), this);
+  HFRecHitgetter_ = edm::GetterOfProducts<edm::SortedCollection<HFRecHit, edm::StrictWeakOrdering<HFRecHit>>>(
+      edm::ProcessMatch("*"), this);
+  HORecHitgetter_ = edm::GetterOfProducts<edm::SortedCollection<HORecHit, edm::StrictWeakOrdering<HORecHit>>>(
+      edm::ProcessMatch("*"), this);
+  callWhenNewProductsRegistered([this](edm::BranchDescription const& bd) {
+    this->HBHERecHitgetter_(bd);
+    this->HFRecHitgetter_(bd);
+    this->HORecHitgetter_(bd);
+  });
```

Comment:
The earlier behavior could be achieved with e.g.
```suggestion
  callWhenNewProductsRegistered([this](edm::BranchDescription const& bd) {
    // in case of EDAliases, consume only the aliased-for original products
    if (bd.isAnyAlias()) return;
    this->HBHERecHitgetter_(bd);
    this->HFRecHitgetter_(bd);
    this->HORecHitgetter_(bd);
  });
```